### PR TITLE
WIP Tables update

### PR DESCRIPTION
**Pull request by sophistifunk**
# Description

WIP / Don't merge

### CODE REVIEW

**Comment by sophistifunk** [comment]
dontlook

**Comment by michaelneale** [comment]
its looking beautiful @sophistifunk 

screen shot 2017-05-02 at 7 00 52 pm

**Comment by i386** [comment]
@michaelneale @sophistifunk agreed!

**Comment by michaelneale** [comment]
so gave it a try - I can't click on any rows of the activity table - so it seems still to early to even look at the ATH. Branches and PRs are using old table style.

**Comment by i386** [comment]
Looking really great @sophistifunk 

Few things:
* Font links in the table should not be Blue as per the mockup
table-design
* Selector font is strange - probably inheriting the font rules from the table? 
selector-original
whimsy 2017-05-08 10-45-37
* Commit lozenge styling is off. Should be aligned right and have ellipsis rules
ios app 2017-05-08 10-48-02
commit-lozenge-original

**Comment by i386** [comment]
@sophistifunk I am happy for this to go in as is to avoid more merge drama but we would need to follow up after that.

**Comment by sophistifunk** [comment]
Gotta fix the Link issue first, once I can figure it out.

**Comment by i386** [comment]
@sophistifunk I've ticketed this feedback up

- https://issues.jenkins-ci.org/browse/JENKINS-44105
- https://issues.jenkins-ci.org/browse/JENKINS-44106
- https://issues.jenkins-ci.org/browse/JENKINS-44107

**Comment by michaelneale** [comment]
Looking good - clicking works now, nice. Wlil look shortly on what small changes the ATH may need.

On :bug: is that if you click refresh on activity screen it will cause the whole page to refresh. This is an anchor embedded in a whole table anchor - so not sure how that is meant to work.

**Comment by michaelneale** [comment]
Also this looks wrong: 

```
href="/blue/organizations/jenkins/my-pipeline/detail/my-pipeline/4/pipeline"
```
in the href for the row - it should be more like: 
```
href="/blue/organizations/jenkins/my-pipeline/detail/runs/4/pipeline"
```
right?

**Comment by sophistifunk** [comment]
> if you click refresh on activity screen it will cause the whole page to refresh

.... I should hope so? 

I'm not sure I get what you mean.

**Comment by michaelneale** [comment]
@sophistifunk oops I mean "re run". Ie re run the build from the activity list (it has a refresh icon).

**Comment by michaelneale** [comment]
@sophistifunk https://github.com/jenkinsci/blueocean-plugin/pull/899 ws merged, as it was ready for a while. 

Unfortunately that creates a conflict: https://github.com/jenkinsci/blueocean-plugin/pull/1028/conflicts - as it ads an extra link back to the branches - is this a big deal?

**Comment by sophistifunk** [comment]
Thought it was a conflict I'd already resolved

**Comment by sophistifunk** [comment]
Green biatches!

**Comment by sophistifunk** [comment]
But now waiting again because of merge

**Comment by michaelneale** [comment]
what a time to be alive

**Comment by sophistifunk** [comment]
Hmm, looks like somebody has added new tests that I've pulled, shouldn't be a big job

**Comment by michaelneale** [comment]
they are the new logged in favourites tests - very nice to see.

**Comment by michaelneale** [comment]
@sophistifunk a small fix from master from cliff that now actually runs the lint tests is needed. Also the ATH is now less shit if you update this to master at your leisure.

**Comment by michaelneale** [comment]
@sophistifunk one thing to note when resolving conflicts - commitMessages.js is dead (it was bad). It is now CommitMessagesTest.java (and vastly smaller and simpler)

**Comment by sophistifunk** [comment]
No worries I've made a note of this in case the merges leave it in my branch

**Comment by michaelneale** [comment]
@sophistifunk how about the other merges - hopefully nothing too major ...

**Comment by michaelneale** [comment]
Looking beautiful. 

The pill for commits is not quite right on the activity tab (branches likely the same too): hope that helps:

screen shot 2017-05-30 at 1 37 39 pm

**Comment by i386** [comment]
@michaelneale @sophistifunk these issues that I raised two weeks ago are still there - looking good though!

- Blue links https://issues.jenkins-ci.org/browse/JENKINS-44106 
- Commit lozenge https://issues.jenkins-ci.org/browse/JENKINS-44107

File: acceptance-tests/src/test/js/multibranch/multibranchOpening.js [hunk 1]
```diff
@@ -98,8 +98,9 @@ module.exports = {
          blueActivityPage.waitForElementVisible('.branches');
          blueActivityPage.click(".branches");
 
-         blueActivityPage.waitForElementVisible('tr[id^="feature"]');
-         blueActivityPage.click('tr[id^="feature"]');
+         const rowSelector = '.JTable-row[data-branch="feature"]'
```
Comment (michaelneale) [inline acceptance-tests/src/test/js/multibranch/multibranchOpening.js]: :bug: needs to be feature/1 - as that is the data value (before it was a partial match).

**Comment by michaelneale** [comment]
@sophistifunk please obtain some help from @imeredith on how to rid us of that annoying test.

**Comment by michaelneale** [comment]
@sophistifunk there was a selector looking for some "actions" class that is no longer there on branches tab - so I correct it... will await further barfing... but getting really good!

**Comment by michaelneale** [comment]
@sophistifunk and it passes!

**Comment by michaelneale** [comment]
@sophistifunk @i386 looking great - I assume PR tab is still to be done? branches look AMAAAAZING

**Comment by sophistifunk** [comment]
GREEN AGAIN

success


@i386 feel like checking this out and giving it a whirl?

**Pull request merged**